src/hextools/germ/caproto_ioc.py

```diff
@@ -96,64 +220,73 @@ async def stage(self, instance, value):
 
             self._h5file_desc = h5py.File(self._data_file, "x", libver="latest")
             group = self._h5file_desc.create_group("/entry")
-            self._frame_shape = self.ophyd_det._frame_shape
+            frame_shape = self.frame_shape.value
             self._dataset = group.create_dataset(
                 "data/data",
-                data=np.full(fill_value=np.nan, shape=(1, *self._frame_shape)),
-                maxshape=(None, *self._frame_shape),
-                chunks=(1, *self._frame_shape),
+                data=np.full(fill_value=np.nan, shape=(1, *frame_shape)),
+                maxshape=(None, *frame_shape),
+                chunks=(1, *frame_shape),
                 dtype="float32",
             )
             self._h5file_desc.swmr_mode = True
             return True
 
-        elif value == "unstaged":
+        if value == StageStates.UNSTAGED.value:
```

Comment:
in this block also reset all of the various internal state to `None`.

It won't "fix" the problem, but will at least convert the errors to ones that are clearer.